include/targets/cuda/kernel.h

```diff
@@ -89,14 +89,19 @@ namespace quda
 
     auto i = threadIdx.x + blockIdx.x * blockDim.x;
     auto j = threadIdx.y + blockIdx.y * blockDim.y;
-    if (j >= arg.threads.y) return;
 
-    while (i < arg.threads.x) {
+    if constexpr (Arg::check_bounds) {
```

Comment:
Should this go into, for ex, the `hip` target for completeness/consistency even if it's unused for now?

Reply:
https://github.com/lattice/quda/pull/1497/commits/6451e7fb56974b902baacdb64f547ef8eb5231c4

Reply:
Actually even for the CUDA target this check_bounds variable is only used in `kernel2D_impl` (i.e., not used in `kernel3D_impl`, etc). Adding similar changes to all the other kernels seems too aggressive though.

Reply:
@maddyscientist you might have a better way of doing this? So currently `check_bounds`/`check_bounds_` are added to `kernel_param`, and it is picked up only here in `Kernel2D_impl`, in the CUDA target.